### PR TITLE
Opt for MIT hecs licensing now that hecs supports it

### PR DESCRIPTION
**Pull request by cart**
A big thanks to upstream hecs / @Ralith for adding support for an MIT license option. This allows us to simplify the bevy_ces licensing story significantly: https://github.com/Ralith/hecs/pull/115

@Ralith let me know if anything looks incorrect to you!

The hecs MIT license now lives in the bevy_ecs/license folder alongside the bevy MIT license

### CODE REVIEW

**Comment by cart** [comment]
@Ralith: i should probably verify that removing ` // Copyright 2019 Google LLC ` comments are ok. I wasn't thinking about that when I removed the apache license boilerplate.

**Comment by Ralith** [comment]
Will report back with input from the lawyers.

**Comment by cart** [comment]
Awesome. Thanks!

File: crates/bevy_ecs/README.md [hunk 1]
```diff
@@ -1,7 +1,3 @@
 # Bevy ECS
 
-Bevy Engine's entity component system
-
-## Licensing
-
-Bevy ECS has its roots in hecs, which is licensed as Apache 2.0. All original hecs code is licensed under Apache 2.0. All added/modified code is dual licensed under MIT and Apache 2.0. Files with an Apache 2.0 license header (with Google LLC as the copyright holder) were from the "original hecs" codebase. Files without the header were created by Bevy contributors.
```
Comment (faith) [inline crates/bevy_ecs/README.md:7]: I dont think you shoulf get rid of this as its still licensed, just under MIT

**Review by faith — CHANGES_REQUESTED** [review]
Its still licensed just under MIT so i am not sure about getting rid of Licensing section. Id say keep but just edit a bit

File: crates/bevy_ecs/README.md (hunk 1 above)
Comment (cart) [reply crates/bevy_ecs/README.md:7]: Thats a good point. I'll re-add something along those lines if/when we get the go-ahead from the lawyers.

**Comment by cart** [comment]
Closing in favor of #1525, which replaces all hecs code in Bevy (except the entity allocator, which was submitted to this repo by the original author)

**Pull request closed without merging**